plugin/v2/plugin.go

```diff
@@ -205,18 +213,53 @@ func (p *Plugin) GetTypes() []types.PluginInterfaceType {
 
 // InitSpec creates an OCI spec from the plugin's config.
 func (p *Plugin) InitSpec(s specs.Spec, libRoot string) (*specs.Spec, error) {
-	rootfs := filepath.Join(libRoot, p.PluginObj.ID, "rootfs")
+	p.Rootfs = filepath.Join(libRoot, p.PluginObj.ID, "rootfs")
 	s.Root = specs.Root{
-		Path:     rootfs,
+		Path:     p.Rootfs,
 		Readonly: false, // TODO: all plugins should be readonly? settable in manifest?
 	}
 
+	// TODO: the following should be only if granting device permissions
+	rwm := "rwm"
+	s.Linux.Resources.Devices = []specs.DeviceCgroup{{Allow: true, Access: &rwm}}
+
 	mounts := append(p.PluginObj.Config.Mounts, types.PluginMount{
 		Source:      &p.RuntimeSourcePath,
 		Destination: defaultPluginRuntimeDestination,
 		Type:        "bind",
 		Options:     []string{"rbind", "rshared"},
 	})
+
+	if p.PluginObj.Manifest.Network.Type != "" {
```

Comment:
Lets look to refactor. At the least, we should perform more error checking such as stating the host file before adding to mounts.